avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1738,12 +1775,37 @@ private static boolean isInstanceOfAvroSchemaTypeForSimpleSchema(Schema fieldSch
     }
     for (Class type : classes) {
       if (type.isInstance(value)) {
-        return true;
+        if (fieldSchema.type() == Schema.Type.BYTES
+            && fieldSchema.parameters() != null
+            && fieldSchema.parameters().containsKey(CONNECT_AVRO_FIXED_SIZE)) {
+          if (fixedValueSizeOf(value)
+              == Integer.parseInt(fieldSchema.parameters().get(CONNECT_AVRO_FIXED_SIZE))) {
+            return true;
+          }
+        } else {
+          return true;
+        }
       }
     }
     return false;
   }
 
+  /**
+   * Get size of bytes value tagged as fixed
+   */
+  private static int fixedValueSizeOf(Object value) {
```

Comment:
Maybe have this return a boolean, so it can apply a different test to `GenericFixed`?